rotkehlchen/api/v1/resources.py

```diff
@@ -1697,6 +1712,10 @@ def post(
             file: FileStorage,
             **kwargs: Any,
     ) -> Response:
+        _, tmpfilepath = tempfile.mkstemp()
+        file.save(tmpfilepath)
+        self.rest_api.import_tmp_files[file] = Path(tmpfilepath)
+
```

Comment:
Questions about this:

1. Why moving it here?
2. if you already know the filepath why not pass the filepath instead of the file? In the lofic seems that you are using the filepath and not the file itself
3. I would change this to use the TempFile library to avoid forgetting about this files and remove the import_tmp_files if this is possible. I don't see why not right now

Reply:
> Why moving it here?

Previously there were 2 methods. `import_data` and `_import_data`. `import_data` (the parent function) had this logic of making a temporary file and then it was asynchronously calling `_import_data`. Now that there is only one function `import_data` which is asynchronous, this logic of creating a temporary file has to be executed before calling `import_data`. I thought that a good place to do it would be `resources.py`

As for whether to pass the path instead of the file directly -- looks reasonable, but I am not sure. This code was made some time ago by @LefterisJP and I don't remember the reason for choosing the current approach.